### PR TITLE
diag: do not emit diagnostic logging items if omsagent version is old

### PR DESCRIPTION
**Pull request by chparimi**
Do not support diagnostic logging if omsagent version is older than
the version 1.3.4-127.

### CODE REVIEW

**Comment by chparimi** [comment]
@Microsoft/omsagent-devs

File: source/code/plugins/oms_diag_lib.rb [hunk 1]
```diff
@@ -126,6 +198,15 @@ def GetCurrentFormattedTimeForDiagLogs()
           Time.now.utc.strftime("%Y-%m-%dT%H:%M:%S.%6NZ")
       end
 
+      # Method used to get major minor version parts as array of omsagent version
+      def GetVersionAsArray(versionStr)
```
Comment (lagalbra) [inline source/code/plugins/oms_diag_lib.rb]: I think this should be a little more explicit as major.minor.patch-build; see our [bundle code](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/bundle_skel.sh#L133) for how we compare version numbers there.

File: source/code/plugins/oms_diag_lib.rb [hunk 2]
```diff
@@ -24,8 +24,78 @@ class Diag
...
+
+        begin
+          # Read installinfo.txt
+          versionline = IO.readlines(@@InstallInfoPath)[0]
+
+          # Extract version number
+          versionnum = versionline.split()[0]
+
+          # Extract major and minor parts of version number
+          major, minor = GetVersionAsArray(versionnum)
```
Comment (lagalbra) [inline source/code/plugins/oms_diag_lib.rb]: See my other comment below; I think this should be return as major, minor, patch, build.

File: source/code/plugins/oms_diag_lib.rb [hunk 3]
```diff
@@ -24,8 +24,78 @@ class Diag
...
+             !minor.is_a?(String)
+            @@IsDiagSupported = false
+            return @@IsDiagSupported
+          end
+
+          # Compare version number
+          min_major, min_minor = GetVersionAsArray(DIAG_MIN_VERSION)
+
+          # Compare major parts
+          for x in 0..2
```
Comment (lagalbra) [inline source/code/plugins/oms_diag_lib.rb]: Once you make the change to consider all 4 parts separately, make sure the logic procedure matches the logic in bundle_skel [here](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/bundle_skel.sh#L159).

File: source/code/plugins/oms_diag_lib.rb [hunk 4]
```diff
@@ -24,8 +24,78 @@ class Diag
...
+      @@InstallInfoPath = "/etc/opt/microsoft/omsagent/sysconf/installinfo.txt"
+
+      # Method to check if diagnostic logging is supported
+      #
+      # Description:
+      # This would tell if current omsagent version supports
+      # diagnostic logging
+      #
+      # NOTE: If this returns false then logs will be rejected silently
+      def IsDiagEnabled()
```
Comment (lagalbra) [inline source/code/plugins/oms_diag_lib.rb]: The method name is a little unclear; why didn't you use IsDiagSupported()?

File: test/code/plugins/oms_diag_lib_test.rb [hunk 5]
```diff
@@ -240,4 +254,95 @@ def test_case_record_proc_01
...
+
+        # Checking that diag is enabled
+        assert_equal(true,
+                     OMS::Diag.IsDiagEnabled(),
+                     "Diagnostic logging should be supported for when version is min version")
+    end
+
+    # Checking diagnostic support with higher omsagent version
+    def test_case_version_check_03
+        versions = ["1.3.4-128", "1.3.5-127", "1.4.4-127", "2.3.4-127"]
```
Comment (lagalbra) [inline test/code/plugins/oms_diag_lib_test.rb]: I would also suggest adding cases like 1.4.0-1, which should evaluate as a higher version than 1.3.4-127, even though the patch and build numbers are lower.

File: source/code/plugins/oms_diag_lib.rb (hunk 2 above)
Comment (chparimi) [inline source/code/plugins/oms_diag_lib.rb]: Ok will do that.

File: source/code/plugins/oms_diag_lib.rb (hunk 3 above)
Comment (chparimi) [inline source/code/plugins/oms_diag_lib.rb]: Checked the logic. It is the same. Just that I would have to make it major, minor patch and build. Will update right away.

File: source/code/plugins/oms_diag_lib.rb (hunk 1 above)
Comment (chparimi) [inline source/code/plugins/oms_diag_lib.rb]: Doing that.

File: test/code/plugins/oms_diag_lib_test.rb (hunk 5 above)
Comment (chparimi) [inline test/code/plugins/oms_diag_lib_test.rb]: Sure will add that.

**Comment by chparimi** [comment]
@lagalbra Can you please approve this PR?

**Pull request merged**